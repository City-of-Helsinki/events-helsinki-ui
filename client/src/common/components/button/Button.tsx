import classNames from "classnames";
import React, { FunctionComponent, ReactElement, SyntheticEvent } from "react";

import styles from "./button.module.scss";

export enum ButtonStyles {
  MEDIUM_PRIMARY = "MediumPrimary"
}

export type ButtonStyleType = ButtonStyles;

type ReactButton = React.ButtonHTMLAttributes<HTMLButtonElement>;

interface Props extends ReactButton {
  buttonStyle: ButtonStyleType;
  className?: string;
  icon?: ReactElement;
  isBlock?: boolean;
  disabled?: boolean;
  onClick?: (event?: SyntheticEvent) => void;
  size?: ButtonStyleType;
}

const Button: FunctionComponent<Props> = ({
  buttonStyle,
  children,
  className,
  disabled,
  icon,
  isBlock,
  onClick,
  type = "button",
  ...rest
}) => {
  return (
    <button
      className={classNames(
        styles.btn,
        {
          [styles[`btn${buttonStyle}`]]: buttonStyle,
          [styles.btnBlock]: isBlock
        },
        className
      )}
      disabled={disabled}
      onClick={onClick}
      type={type}
      {...rest}
    >
      {icon}
      {children}
    </button>
  );
};

export default Button;
