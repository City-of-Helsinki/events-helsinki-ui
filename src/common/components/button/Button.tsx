import classNames from "classnames";
import { Button as HdsButton, ButtonProps } from "hds-react";
import React from "react";

import styles from "./button.module.scss";

const Button: React.FC<ButtonProps> = ({
  children,
  color = "primary",
  disabled,
  ...rest
}) => {
  return (
    <HdsButton
      className={classNames(styles[color], { [styles.disabled]: disabled })}
      color={color}
      disabled={disabled}
      {...rest}
    >
      {children}
    </HdsButton>
  );
};

export default Button;
