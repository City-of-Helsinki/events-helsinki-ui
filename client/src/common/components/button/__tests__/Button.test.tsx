import * as React from "react";
import renderer from "react-test-renderer";

import Button from "../Button";

test("Button matches snapshot", () => {
  const component = renderer.create(
    <Button color="primary" size="md">
      Test
    </Button>
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

export {};
