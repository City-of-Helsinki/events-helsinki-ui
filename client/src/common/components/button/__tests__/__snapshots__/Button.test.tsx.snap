// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button matches snapshot 1`] = `
<button
  className="btn mdSize primary"
  type="button"
>
  Test
</button>
`;
